### PR TITLE
Adds pypi CI

### PR DESCRIPTION
**Pull request by oke-aditya**
Adds CI for packaging and releasing to PyPI.

This will need #243 to be merged before. As that will ensure if packaging is alright. Then we can proceed for this.

Helps #209 #182 #157 .

### CODE REVIEW

**Comment by oke-aditya** [comment]
This too is ready to merge @lgvaz . Checks pass as suggested. We need to do trial run checks of the release.

File: examples/training.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 # Create both training and validation dataloaders
 train_dl = faster_rcnn.train_dataloader(train_ds, batch_size=16, num_workers=4, shuffle=True)
+
```
Comment (lgvaz) [inline examples/training.py:33]: Can you rollback this file? This update in unnecessary

**Comment by lgvaz** [comment]
This one I still need some more time to fully understand what is happening

File: examples/training.py (hunk 1 above)
Comment (oke-aditya) [reply examples/training.py:33]: Yes rolled it back. Ctrl + Z is magic.

**Comment by oke-aditya** [comment]
@lgvaz 
Now that we fully understand how it works. We can merge it. And try it. I have tried this CI action to deploy package. It's pretty simple

File: .github/workflows/pypi-release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: Install dependencies
+      run: >-
+        python -m pip install --user --upgrade setuptools wheel
+    - name: Build
+      run: >-
+        python setup.py sdist bdist_wheel
+    # We do this, since failures on test.pypi aren't that bad
+    - name: Publish to Test PyPI
+      if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
```
Comment (lgvaz) [inline .github/workflows/pypi-release.yml:34]: Can you explain this line?

Why we publish if tagged **or** release? Is publishing with the tag a pre-release?

**Comment by lgvaz** [comment]
Can you give some usage examples?

Give us a piece of code on how to correctly tag a commit for this to work, what will be shown in Pypi, how to tag pre-release vs release.

**Comment by oke-aditya** [comment]
@lgvaz I was a bit away for few days. (Univ exams, recruitements etc). I will add these stuff and show you a demo how this works in a meet maybe.

**Comment by lgvaz** [comment]
Cool, don't worry, I'll this additional info directly here then =)

**Pull request merged**